### ISSUE TITLE
HL2 Spt wont load

### ISSUE THREAD

**Issue by Efuri** [YaLTeR/SourcePauseTool#42]
Hl2 old engine (BitRain)
So im using old engine half life 2 and i put the spt.dll file in bin file but when trying to load the plugin it says 
Could not get IServerPluginCallbacks interface from plugin "spt"Unable to load plugin "spt"

**Comment by YaLTeR** [comment]
You need the `spt-oe.dll` for OE.

**Comment by Arna57** [comment]
where exacly do i put the files? ive tried spt-oe in the folder with hl2, putting hl2 and spt inside bin, hl2 inside bin and oe outside, the reverse, and all i get is https://gyazo.com/1c5e81f0fa79deb6262b8ec218c9765f
ive also done the -insecure launch parameter but im geniuenly stuck now

**Comment by PackSciences** [comment]
OE => spt-oe.dll
Moving "HL2" No
The only manoeuvre you need to do is download spt-oe.dll into the bin folder inside hl2 folder : "For older engines you will need to place spt.dll into the topmost bin directory (the one with a lot of different DLL files including engine.dll, not the one with client.dll and server.dll)." - Vladimir YaLTeR Putin
Finally, plugin_load spt-oe as you did in your screenshot.

**Comment by mikesmiffy128** [comment]
*I'm just ignoring what pack just said because I was typing this when he posted it and what he said doesn't make any sense...*

@Arna57 I don't know why you'd ever try to move hl2 around (what does that even mean, the exe, the directory, what...). Nowhere does anything suggest you should do that, including basic common sense, one would hope. Hopefully you've worked out how to put it back the way it was.

Here's what the directory layout would generally look like:
```
/
    hl2/
        bin/
        cfg/
        Other things
    bin/
        spt-oe.dll
        Other things
    hl2.exe
    Some other things
```

In other words, the top-level bin directory should have spt-oe.dll, or the directory that exists beside hl2.exe. Now there is a slight issue with this if you're on the 4044 Steam mod version of the game. In this case the hl2.exe is under `SteamApps/common/Source SDK Base 2006` or something thereabouts, I can't exactly remember. There's still a bin folder next to that exe, so again that's where you would put the DLL.

Hopefully this makes sense... and hopefully I didn't completely get it wrong, because then I'd have to worry about whether my memory is working correctly.

**Comment by PackSciences** [comment]
What I originally said was The only manoeuvre you need to do is download spt-oe.dll into the bin folder inside hl2 folder : "For older engines you will need to place spt.dll into the topmost bin directory (the one with a lot of different DLL files including engine.dll, not the one with client.dll and server.dll)."
But then I realized I meant inside the HL2 OE folder (the one inside sourcemods) and not like HL2 inside HL2 OE, which was confusing, so I edited my post.

**Comment by mikesmiffy128** [comment]
That's still confusing... what I wrote above should work fine.

**Comment by PackSciences** [comment]
Oh ok :'(

**Comment by Centaurium** [comment]
ok, so the short answer is this: 
1. Download spt-oe.dll. (you might need this version if features don't work: https://cdn.discordapp.com/attachments/238860179855114241/401201893658853387/spt-oe.dll)
2. Navigate to SteamApps/common/Source SDK Base 2006/bin, and move spt-oe.dll into that folder
3. Launch OE, "plugin_load spt-oe" (if this command already exists in your autoexec.cfg, don't enter this in twice).

To mike and pack, I don't know why this had to be so complicated when you could've answered his question in 3 lines.
